File: 62.unique-paths.java
/*
 * @lc app=leetcode id=62 lang=java
 *
 * [62] Unique Paths
 */

// @lc code=start
class Solution {
    public int uniquePaths(int m, int n) {

     int[][] matrix_paths = new int[m][n];

        for (int row = 0; row < m; row++) {
            for (int col = 0; col < n; col++) {
                
                if (row == 0 || col == 0) {
                    matrix_paths[row][col] = 1;
                } else {
                    matrix_paths[row][col] = matrix_paths[row - 1][col] + matrix_paths[row][col - 1];
                }
                
            }
        }

        return matrix_paths[m - 1][n - 1];
    }
}
// @lc code=end

File: 62.unique-paths.py
#
# @lc app=leetcode id=62 lang=python3
#
# [62] Unique Paths
#

# @lc code=start
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        paths_matrix = [[1] * n] * m

        for row in range(1, m):
            for col in range(1, n):
                paths_matrix[row][col] = paths_matrix[row - 1][col] + paths_matrix[row][col - 1]
        
        return paths_matrix[m - 1][n - 1]
# @lc code=end

